neuralpp/symbolic/interval.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+from abc import abstractmethod
+from typing import Iterable, Callable
+from constants import if_then_else
+
+from .expression import Variable, Expression, Context
+from .z3_expression import Z3SolverExpression
+
+
+class ClosedInterval:
+    """ [lower_bound, upper_bound] """
+    @property
+    @abstractmethod
```

Comment:
Should these be abstract methods?

Reply:
No, we can just implement this class. It's just that I want to write some legit code in a PR.